openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIQueryParameterMapper.java

```diff
@@ -0,0 +1,249 @@
+            // Handle required array type query parameter
+            ArrayTypeDescriptorNode arrayNode = (ArrayTypeDescriptorNode) queryParam.typeName();
+            return handleArrayTypeQueryParameter(queryParamName, arrayNode);
+        } else {
+            QueryParameter queryParameter = createContentTypeForMapJson(queryParamName, false);
+            if (!apidocs.isEmpty() && queryParam.paramName().isPresent() && apidocs.containsKey(queryParamName)) {
+                queryParameter.setDescription(apidocs.get(queryParamName.trim()));
+            }
+            return queryParameter;
+        }
+    }
+
+    /**
+     * Create OAS query parameter for default query parameters.
+     */
+    public Parameter createQueryParameter(DefaultableParameterNode defaultableQueryParam) {
+
+        String queryParamName = defaultableQueryParam.paramName().get().text();
+        boolean isQuery = !defaultableQueryParam.paramName().get().text().equals(Constants.PATH) &&
+                defaultableQueryParam.annotations().isEmpty();
+
+        QueryParameter queryParameter = new QueryParameter();
+        if (defaultableQueryParam.typeName() instanceof BuiltinSimpleNameReferenceNode && isQuery) {
+            queryParameter.setName(queryParamName);
+            Schema openApiSchema = ConverterCommonUtils.getOpenApiSchema(
+                    defaultableQueryParam.typeName().toString().trim());
+            queryParameter.setSchema(openApiSchema);
+            if (!apidocs.isEmpty() && defaultableQueryParam.paramName().isPresent() &&
+                    apidocs.containsKey(queryParamName)) {
+                queryParameter.setDescription(apidocs.get(queryParamName.trim()));
+            }
+        } else if (defaultableQueryParam.typeName().kind() == OPTIONAL_TYPE_DESC && isQuery) {
+            // Handle optional query parameter
+            queryParameter = setOptionalQueryParameter(queryParamName,
+                    ((OptionalTypeDescriptorNode) defaultableQueryParam.typeName()),
+                    Constants.TRUE);
+        } else if (defaultableQueryParam.typeName() instanceof ArrayTypeDescriptorNode && isQuery) {
+            // Handle required array type query parameter
+            ArrayTypeDescriptorNode arrayNode = (ArrayTypeDescriptorNode) defaultableQueryParam.typeName();
+            queryParameter = handleArrayTypeQueryParameter(queryParamName, arrayNode);
+        } else {
+            queryParameter = createContentTypeForMapJson(queryParamName, false);
+            if (!apidocs.isEmpty() && defaultableQueryParam.paramName().isPresent() &&
+                    apidocs.containsKey(queryParamName)) {
+                queryParameter.setDescription(apidocs.get(queryParamName.trim()));
+            }
+        }
+
+        SyntaxKind[] validExpressionKind = {STRING_LITERAL, NUMERIC_LITERAL, BOOLEAN_LITERAL, LIST_CONSTRUCTOR,
+                NIL_LITERAL, MAPPING_CONSTRUCTOR};
```

Comment:
Shall we move these valid expression kinds to the class level, instead of creating new arrays in each method call?